src/graph/graph.c

```diff
@@ -258,14 +268,15 @@ void _MatrixSynchronize(const Graph *g, RG_Matrix rg_matrix) {
 
 	// If the matrix has pending operations or requires
 	// a resize, enter critical section.
-	if(pending || (n_rows != dims) || (n_cols != dims)) {
+	if(pending || require_resize) {
 		// Double-check if resize is necessary.
 		GrB_Matrix_nrows(&n_rows, m);
 		GrB_Matrix_ncols(&n_cols, m);
 		dims = Graph_RequiredMatrixDim(g);
-		if((n_rows != dims) || (n_cols != dims)) {
-			assert(GxB_Matrix_resize(m, dims, dims) == GrB_SUCCESS);
-		}
+		require_resize = ((n_rows < dims || n_cols < dims) ||
```

Comment:
can `require_resize` change from `true` to `false` or vise versa before and after matrix locking? 
TBH, I don't recall why we need second validation for that?

Reply:
It's possible for two threads to try to retrieve the same matrix at the same time, in which case only one should actually update the matrix. As such, we double-check that the resize is necessary after lock acquisition, in case we're the second thread in this scenario.